internal/host/host.go

```diff
@@ -207,6 +207,47 @@ func (m *Manager) HandleInstallationFailure(ctx context.Context, h *models.Host)
 	return err
 }
 
+// populateDisksEligibility updates an inventory json string by updating the eligibility
+// struct of each disk in the inventory with service-side checks for disk eligibility, in
+// addition to agent-side checks that have already been performed. The reason that some
+// checks are performed by the agent (and not the service) is because the agent has data
+// that is not available in the service.
+func (m *Manager) populateDisksEligibility(inventoryString string) (string, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(inventoryString), &inventory); err != nil {
+		return "", err
+	}
+
+	for _, disk := range inventory.Disks {
+		if !disk.InstallationEligibility.Eligible && len(disk.InstallationEligibility.NotEligibleReasons) == 0 {
```

Comment:
This would probably not need a comment if you make it a function on disk and call it (`validEligibility` or something).
Also since we do this same check [elsewhere](https://github.com/openshift/assisted-service/pull/804/files#diff-4c707383b7985c70d26467ad3b9a6cb21aeebf6208d3fbdd4997da55c5d6b30bR96) it would be good to put this logic in one place in case it needs to be updated in the future.